embed-code-go/embedding/parsing/context.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2024, TeamDev. All rights reserved.
+//
+// Redistribution and use in source and/or binary forms, with or without
+// modification, must retain the above copyright notice and the following
+// disclaimer.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+package parsing
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+)
+
+// EmbeddingInParsingContext contains the information about the position in the source and the
+// resulting Markdown files.
+//
+// Embedding - an Instruction, containing all the needed embedding information.
+//
+// SourceStartIndex - an index of the StartState line in the original markdown file.
+//
+// SourceEndIndex - an index of the end line in the original markdown file.
+//
+// ResultStartIndex - an index of the StartState line in the result markdown file.
+//
+// ResultEndIndex - an index of the end line in the result markdown file.
+type EmbeddingInParsingContext struct {
+	Embedding        Instruction
+	SourceStartIndex int
+	SourceEndIndex   int
+	ResultStartIndex int
+	ResultEndIndex   int
+}
+
+// Context represents the context for parsing a file containing code embeddings.
+//
+// Embedding - a pointer to the embedding instruction.
+//
+// Source - a list of strings representing the original markdown file.
+//
+// MarkdownFilePath - a path to the markdown file.
+//
+// LineIndex - an index of the current line in the markdown file.
+//
+// Result - a list of strings representing the markdown file updated with embedding.
+//
+// CodeFenceStarted - a flag indicating whether a code fence has been started.
+//
+// CodeFenceIndentation - an indentation of the markdown's code fences.
+//
+// FileContainsEmbedding - a flag indicating whether the file contains an embedding instruction.
+//
+// Embeddings - a list of embedding instructions found in the markdown file.
+//
+// EmbeddingsNotFound - a list of embedding instructions that are not found in the code.
+//
+// UnacceptedEmbeddings - a list of embedding instructions that are not accepted by the parser.
+type Context struct {
```

Comment:
Currently there is two fields `Embedding` and  `Embeddings` they have different types, by the description and field name it should be the same type, or at least there should be more details in the doc of the `Embeddings`. 

Another issue with this field, maybe I missed something but I can't find any usages of `Embedding` except `.Content` and `nil` check. Just a proposal: probably we can replace it with some `content` with the corresponding doc.

Reply:
I found that this is resolved in the following [PR](https://github.com/SpineEventEngine/embed-code/pull/54) please ignore this comment.